crates/matrix-sdk/src/sliding_sync/list/mod.rs

```diff
@@ -971,38 +972,38 @@ mod tests {
         {
             let list = SlidingSyncList::builder("foo")
                 .sync_mode(SlidingSyncMode::Selective)
-                .ranges(ranges![(0, 1)].to_vec())
+                .ranges(vec![0u32..=1])
                 .build();
 
             {
                 let lock = list.inner.ranges.read().unwrap();
                 let ranges = Observable::get(&lock);
 
-                assert_eq!(ranges, &ranges![(0, 1)]);
+                assert_eq!(ranges, &vec![0..=1]);
             }
 
-            list.add_range((2u32, 3)).unwrap();
+            list.add_range(2u32..=3).unwrap();
```

Comment:
I start wondering if it's mandatory to explicit the type `u64` or `u32` here? Rust may be able to infer it automatically, right?

Reply:
Unfortunately, it really wants the integer to be a `i32` (probably because it infers first the type for the range ctor before determining how the range is used), so the integer type has to be explicited :pensive: